backend/backends/stdout/stdout.go

```diff
@@ -79,14 +79,14 @@ func (client *Client) SendMetrics(metrics types.MetricMap) error {
 		fmt.Fprintf(buf, "stats.set.%s %d %d\n", nk, len(set.Values), now)
 	})
 
-	fmt.Fprintf(buf, "statsd.numStats %d %d\n", metrics.NumStats, now)
-	fmt.Fprintf(buf, "statsd.processingTime %f %d\n", float64(metrics.ProcessingTime)/float64(time.Millisecond), now)
-
-	_, err := buf.WriteTo(log.StandardLogger().Writer())
-	if err != nil {
-		return err
-	}
-	return nil
+	writer := log.StandardLogger().Writer()
+	defer func() {
+		if err := writer.Close(); err != nil && retErr == nil {
```

Comment:
In the past I've been burnt many times by code that was ignoring errors (even logs about "not important" errors could be very useful to troubleshoot something). It is better to find out about any errors happening than see clean logs and think that all is good. I.e. we can always silence down unnecessary noise later but we will never know that something is wrong if we ignore errors from start.

If you think about it, there must be a reason the OS/C-level functions for I/O (including closing the file) return error. If all such errors were not important or non-existent close() would not return error. The actual reason is when you close the file/stream/socket OS flushes the data to disk/network if some was buffered, syncs meta information to disk about last file update time/etc. So if a error occurs while closing a file it might mean e.g. that HDD just died. Don't we want to find out about such errors?

Also you might be interested in reading comments here https://github.com/google/guava/issues/1118